src/models/StyledComponent.js

```diff
@@ -16,138 +16,186 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
   /* We depend on components having unique IDs */
   const identifiers = {}
   const generateId = (_displayName: string) => {
-    const displayName = _displayName
-      .replace(/[[\].#*$><+~=|^:(),"'`]/g, '-') // Replace all possible CSS selectors
-      .replace(/--+/g, '-') // Replace multiple -- with single -
+    const displayName = typeof _displayName !== 'string' ?
+      'sc' : _displayName
+        .replace(/[[\].#*$><+~=|^:(),"'`]/g, '-') // Replace all possible CSS selectors
```

Comment:
Thanks, that shouldn't change a lot, but let's make every bit count :)

https://jsperf.com/regexp-indexof-perf